python/packages/autogen-core/src/autogen_core/components/_routed_agent.py

```diff
@@ -47,13 +47,19 @@ class MessageHandler(Protocol[AgentT, ReceivesT, ProducesT]):  # type: ignore
     async def __call__(agent_instance: AgentT, message: ReceivesT, ctx: MessageContext) -> ProducesT: ...
 
 
-# NOTE: this works on concrete types and not inheritance
-# TODO: Use a protocol for the outer function to check checked arg names
```

Comment:
@jackgerrits @ekzhu  I addressed this TODO but we're still facing same issue. Any guidance you might be able to provide?